odl/util/ufuncs.py

```diff
@@ -68,63 +70,50 @@
 """.format(name)
     UFUNCS.append((name, n_in, n_out, doc))
 
-RAW_REDUCTIONS = [('sum', 'sum', 'Sum of array elements.'),
-                  ('prod', 'prod', 'Product of array elements.'),
-                  ('min', 'amin', 'Minimum value in array.'),
-                  ('max', 'amax', 'Maximum value in array.')]
 
-REDUCTIONS = []
-for name, numpyname, descr in RAW_REDUCTIONS:
-    doc = descr + """
-
-See Also
---------
-numpy.{}
-""".format(numpyname)
-    REDUCTIONS += [(name, doc)]
+DEPRECATION_MESSAGE = """
+The `vector.ufuncs` interface is deprecated and will be removed when Numpy
+1.13 becomes the minimum version required by ODL. Use Numpy ufuncs directly,
+e.g., `numpy.sqrt(x)` instead of `x.ufuncs.sqrt()`.
+"""
 
 
 # Wrap all numpy ufuncs
 
 def wrap_ufunc_base(name, n_in, n_out, doc):
     """Add ufunc methods to `NtuplesBaseUfuncs`."""
-    wrapped = getattr(np, name)
+    ufunc = getattr(np, name)
     if n_in == 1:
-        if n_out == 0:
-            def wrapper(self):
-                return wrapped(self.vector)
-
-        elif n_out == 1:
+        if n_out == 1:
             def wrapper(self, out=None):
-                if out is None:
-                    out = self.vector.space.element()
-
-                out[:] = wrapped(self.vector)
-                return out
+                if parse_version(np.__version__) >= parse_version('1.13'):
```

Comment:
Okay, maybe not. An alternative transition plan is to fully support this interface until Numpy 1.13 becomes minimum, then add a deprecation warning and remove it in the release after that. This also gives us some leeway to port our own code.